tests/preprocessing/test_data.py

```diff
@@ -471,3 +471,109 @@ def test_inverse_transform(self):
+        assert_estimator_equal(a.fit(df), a.fit(df.compute().values))
+        assert_estimator_equal(a.fit(df.values), a.fit(df.compute().values))
+        assert_estimator_equal(a.fit(df), b.fit(df.compute()))
+        assert_estimator_equal(a.fit(df), b.fit(df.compute().values))
+
+    def test_estimator_comparison(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        a.fit_transform(X).compute()
+        b.fit_transform(X.compute())
+        assert_estimator_equal(a, b)
+
+    def test_array_transform(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        res_a = a.fit_transform(X).compute()
+        res_b = b.fit_transform(X.compute())
+        assert_eq_ar(res_a, res_b)
+
+    def test_transform_array(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        # pass numpy array to fit_transform
+        res_a1 = a.fit_transform(X.compute())
+        # pass dask array to fit_transform
+        res_a2 = a.fit_transform(X).compute()
+        res_b = b.fit_transform(X.compute())
+        assert_eq_ar(res_a1, res_b)
+        assert_eq_ar(res_a2, res_b)
+
+    def test_df_transform(self):
+        a = dpp.PolynomialFeatures(preserve_dataframe=True)
+        b = spp.PolynomialFeatures()
+
+        # pandas dataframe
+        res_pdf = a.fit_transform(df.compute()).values
+        # spp returns a numpy array
+        res_b = b.fit_transform(df.compute())
+        assert_eq_ar(res_pdf, res_b)
+
+    def test_transformed_shape(self):
+        # checks if the transformed objects have the correct columns
+        a = dpp.PolynomialFeatures()
+        a.fit(X)
+        n_cols = len(a.get_feature_names())
+        # dask array
+        assert a.transform(X).shape[1] == n_cols
+        # numpy array
+        assert a.transform(X.compute()).shape[1] == n_cols
+        # dask dataframe
+        assert a.transform(df).shape[1] == n_cols
+        # pandas dataframe
+        assert a.transform(df.compute()).shape[1] == n_cols
+        X_nan_rows = df.values
+        df_none_divisions = X_nan_rows.to_dask_dataframe(columns=df.columns)
+        # dask array with nan rows
+        assert a.transform(X_nan_rows).shape[1] == n_cols
+        # dask data frame with nan rows
+        assert a.transform(df_none_divisions).shape[1] == n_cols
+
+    def test_daskdf_transform(self):
+        a = dpp.PolynomialFeatures(preserve_dataframe=True)
+        b = spp.PolynomialFeatures()
+
+        # dask dataframe
+        trans_dask_df = a.fit_transform(df)
+
+        # pandas dataframe
+        res_pandas_df = a.fit_transform(df.compute())
+        # spp returns a numpy array
+        res_b = b.fit_transform(df.compute())
+        assert_eq_ar(trans_dask_df.compute().values, res_b)
+        assert_eq_df(trans_dask_df.compute().reset_index(drop=True), res_pandas_df)
+
+    def test_df_dont_preserve_df(self):
```

Comment:
This test feels redundant. I believe that everything here is covered elsewhere.

Reply:
It feels redundant, but actually it's not. So far I haven't considered a df as an input with an array as an output. An earlier implementation was not having that, as I was always returning a dataframe if the input was also a dataframe. I'll just add those tests in the `test_df_transform`if I'm managing to duskily that. But maybe it's too cautious.